### PR TITLE
Adds basic end to end test

### PR DESCRIPTION
**Pull request by dk1027**
# Description
This PR adds some basic tests by using a test website that loads analytics.js, clicks around, and captures the network traffic and compares cookies / local storage values.

To run the tests, run `make test-e2e` in the repo root directory:
Should see output like this:
image

Note that the write key is not sensitive data.

When the test run successfully, it captures HTTP requests and saves it under `./test-e2e/staging` directory. In future PRs we will compare these captured requests against a reference.

# Test plan
Test not required because this does not change the ajs library. It only adds tests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,7 @@ package-lock.json
 junit-reports
 
 build/
-.DS_Store
\ No newline at end of file
+.DS_Store
+
+test-e2e/output
+test-e2e/staging
```
Comment (juliofarah) [inline .gitignore:10]:
```suggestion
test-e2e/staging

```

File: test-e2e/ajs.test.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Feature('AJS Bundle');
+
+const assert = require('assert');
+const testSite = 'https://www.library-test-site.com';
+const testWriteKey = 'TEWEu8XrcMVejk8GOulbEx7rHGyuuijV';
```
Comment (juliofarah) [inline test-e2e/ajs.test.ts:5]: note for our future selves:
It'd be cool to have another `Feature` where we could parameterize `testSite` and `testWriteKey` and compare the `.har` generated requests/cookies/local storage against customers websites

File: .gitignore (hunk 1 above)
Comment (dk1027) [reply .gitignore:10]: Both directories should be ignored: codeceptjs dumps errors and captured traffic in `output`, but it has a random suffix, so we move the captured requests into `staging` and renamed them to a canonical name.

Comment (juliofarah) [reply .gitignore:10]: oh no, I got that
this suggestion is just an empty line at the end of the file

**Pull request merged**